client/retry/backoff.go

```diff
@@ -0,0 +1,166 @@
+// Copyright 2023 TiKV Project Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package retry
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/opentracing/opentracing-go"
+	"github.com/pingcap/log"
+	"github.com/pkg/errors"
+	"go.uber.org/zap"
+)
+
+// Backoffer is a utility for retrying queries.
+type Backoffer struct {
+	ctx context.Context
+
+	fn         map[string]backoffFn
+	maxSleep   int
+	totalSleep int
+
+	errors         []error
+	configs        []*Config
+	backoffSleepMS map[string]int
+	backoffTimes   map[string]int
+}
+
+// NewBackoffer (Deprecated) creates a Backoffer with maximum sleep time(in ms).
+func NewBackoffer(ctx context.Context, maxSleep int) *Backoffer {
```

Comment:
There is no need for such a fully functional `backoff`. This is only used when reconnecting members, which is different from the scope of use of `client-go`.  make it simple easy to maintain and readable.

Reply:
fixed